### PR TITLE
Move all Dotenv variables to Mix.Config

### LINKED ISSUE #111: Use config/*.exs instead of dotenv
With reference to the discussion at #109.

### PR DESCRIPTION
**Pull request by ningyuansg**
Resolves #111.

### Features

- There are now only two files that hold secrets,
    - `config/secrets.exs`, which holds secrets that are useful for development
    - `config/prod.secrets.exs`, which holds secrets that are useful only for production
- Each of these files have a corresponding `config/*.example` file.
- With this, the `.env.example` and `Dotenv` file and dependency has been removed.
    - The task `digest` has also been removed
    - The task `server` has been updated to remove its use of Dotenv
- The interval for `Cadet.Updater.Public` has been decreased from 5 minutes to 1 minute.
- The README and moduledocs has been updated accordingly.
- Travis has been updated with an extra `before_install` step to avoid errors w.r.t. config/secrets.exs not existing for `$ mix local.rebar`.

### CODE REVIEW

**Comment by ningyuansg** [comment]
Fixing tests, please hold.

File: README.md [hunk 1]
```diff
@@ -16,47 +16,51 @@ Cadet is the web application powering Source Academy.
 
 ### Setting Up Local Development Environment
 
-Install Elixir dependencies
+1. Install Elixir dependencies
+```
```
Comment (indocomsoft) [inline README.md]: Use ```bash instead

File: README.md [hunk 2]
```diff
@@ -16,47 +16,51 @@ Cadet is the web application powering Source Academy.
...
+$ mix deps.get
+```
 
-    mix deps.get
+2. Initialise development database
+```
```
Comment (indocomsoft) [inline README.md]: Use ```bash

File: README.md [hunk 3]
```diff
@@ -16,47 +16,51 @@ Cadet is the web application powering Source Academy.
...
+$ mix ecto.setup
+```
 
-Initialise development database
+3. Setup the developement secrets (replace the values appropriately)
+```
```
Comment (indocomsoft) [inline README.md]: Use ```bash

File: README.md [hunk 4]
```diff
@@ -16,47 +16,51 @@ Cadet is the web application powering Source Academy.
...
+   also be accessed at http://localhost:4000/swagger.
 
-    IVLE_KEY=your_ivle_lapi_key
 
-If available, also replace the values for GUEST\_USER and GUEST\_PASSWORD
+### Obtaining `access_token` in dev environment
 
-Run the server in your local machine
+You can obtain `access_token` JWT of a user with a given role by simply running:
+
```
Comment (indocomsoft) [inline README.md]: You can enclose this in ```bash as well

File: README.md [hunk 5]
```diff
@@ -16,47 +16,51 @@ Cadet is the web application powering Source Academy.
...
+    mix cadet.token <role>
 
-    mix cadet.server
+For more information, run
```
Comment (indocomsoft) [inline README.md]: And this

**Comment by indocomsoft** [comment]
Oh, and rebase on/merge from `master`, just added `mix format --check-formatted` to travis on `master`

**Comment by ningyuansg** [comment]
Done and fixed.

**Pull request merged**